client/src/utils/CastVoteRecordFiles.ts

```diff
@@ -176,6 +178,9 @@ export default class CastVoteRecordFiles {
         fileCastVoteRecords.map((cvr) => cvr._precinctId)
       )
 
+      const newCastVoteRecords = this.allCastVoteRecords
+      newCastVoteRecords.push(fileCastVoteRecords)
```

Comment:
Prior to this change, this class kept its data structures immutable. I'm assuming that's where some/all of the inefficiency originates. However, the class still presents itself as an immutable data structure but no longer acts that way. Any changes made to an instance derived from a previous one will also mutate that previous one. This may not cause issues for us now, but it's a potential footgun.

Reply:
agreed, but I think we need to rethink this more deeply anyways.